File: 0456-132-pattern/0456-132-pattern.py
class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        stack=[]
        currentMin=nums[0]
        for i in range(1,len(nums)):
            while stack and nums[i]>=stack[-1][0]:
                stack.pop()
            if stack and nums[i]>stack[-1][1]:
                return True
            
            stack.append([nums[i],currentMin])
            currentMin=min(currentMin,nums[i])
        return False
        

File: 0456-132-pattern/0456-132-pattern.java
class Solution {
    public boolean find132pattern(int[] nums) {
        Stack<int[]> stack = new Stack();
        int currentMin = nums[0];
        for (int i = 1; i < nums.length; i++) {
            while (!stack.isEmpty() && nums[i] >= stack.peek()[0]) {
                stack.pop();
            }
            if (!stack.isEmpty() && nums[i] > stack.peek()[1]) {
                return true;
            }
            stack.push(new int[]{nums[i], currentMin});
            currentMin = Math.min(currentMin, nums[i]);
        }
        return false;
    }
}